fdbserver/ptxn/TLogPeekCursor.actor.cpp

```diff
@@ -60,34 +65,34 @@ struct PeekRemoteContext {
 	const std::vector<TLogInterfaceBase*>& pTLogInterfaces;
 
 	// Deserializer
-	SubsequencedMessageDeserializer* pDeserializer;
+	SubsequencedMessageDeserializer& deserializer;
 
 	// Deserializer iterator
-	SubsequencedMessageDeserializer::iterator* pDeserializerIterator;
+	ptxn::details::ArenaWrapper<SubsequencedMessageDeserializer::iterator>& wrappedDeserializerIter;
+
+	// The pointer to the work arena. The work arena stores the data from TLogPeekReply
+	Arena* pWorkArena;
 
 	// If not null, attach the arena in the reply to this arena, so the mutations will still be
 	// accessible even the deserializer gets destructed.
 	Arena* pAttachArena;
 
-	Arena* pWorkArena;
-
 	PeekRemoteContext(const Optional<UID>& debugID_,
```

Comment:
This constructor is confusing as two arenas are needed. So it's worth documenting the meaning of them. Ideally, only one is needed.